kedro/pipeline/modular_pipeline.py

```diff
@@ -52,7 +53,7 @@ def _validate_inputs_outputs(
     free_inputs = {_strip_transcoding(i) for i in pipe.inputs()}
 
     if not inputs <= free_inputs:
-        raise ModularPipelineError("Inputs should be free inputs to the pipeline")
+        raise ModularPipelineError("Inputs must not be outputs from another node")
 
     if outputs & free_inputs:
         raise ModularPipelineError("Outputs can't contain free inputs to the pipeline")
```

Comment:
I'm conscious that we only really define this term deep in [this docstring](https://docs.kedro.org/en/stable/api/kedro.pipeline.Pipeline.html#kedro.pipeline.Pipeline.inputs). 

Perhaps we need to also define the term something like this:

```suggestion
        raise ModularPipelineError("Outputs can't contain free inputs to the pipeline (i.e. The names of inputs that must be provided at runtime so that the pipeline is runnable).
--
 

<br class="Apple-interchange-newline">)")
```

Reply:
I went with "All outputs must be generated by some node in the pipeline" in a bid to keep the message short - what do you think?

Reply:
Yes works well, maybe say 'some node within' rather than 'some node in'?